### DISCUSSION TITLE
RSI/Price Divergence

### DISCUSSION THREAD

**Discussion by jimtollan** [DaveSkender/Stock.Indicators#946] [Ideas]
Hi Dave (et al) - been looking into this and have a gut feel that the indicators library could give a great head start on identifying divergence across trading periods (on RSI and Price).

The question is, how would one set about looking for the outliers that represent this phenomenon. I had thought of using a couple of the methods to identify this, namely: Slope and Linear regression. So far, so good. However, the metal block that I'm hitting is how one should best look for the initiating factors at play. 

My naïve thinking is that you continually look for trend channels on either Price or RSI and when you identify one, you use the same time range (+/- ??) to see if there's a divergence on the partner data.

It's not for lack of thinking that I haven't achieved anything meaningful on this yet, merely that I don't want to reinvent the wheel nor fall into a maze of traps and dead ends that could have been avoided due to opening up dialog on the topic.

If it turns out that I can chain two indicators together to derive this signal, then hurrah!! 😸 

Anyway, thoughts welcome!!
jim

**Comment by DaveSkender** [comment]
See prior conversations for some explorations:

- #434 was a failed attempt to make an indicator out of this concept, but it did inspire some newer enabling features
- #906 explains the possible use of [Pivots](https://dotnet.stockindicators.dev/indicators/Pivots/#content) and newer chaining features

And as you point out, [Slope and Linear Regression](https://dotnet.stockindicators.dev/indicators/Slope/#content) could be an obvious choice to find opposite positive / negative directions.  Adding a trending characteristic condition would potentially strengthen the signal.

```csharp
var priceSlope = quotes
    .GetSlope(..);

var rsiSlope = quotes
    .GetRsi(..)
    .GetSlope(..);

var adxTrend = quotes
    .GetAdx(..);
```

Using a larger or smaller lookback period on these would adjust your confidence vs. signal timing tradeoff.

**Reply by jimtollan** [reply]
cheers dave - from reading thro the links, it would appear that attempts to reliably model this provided as many negatives as it did positives. will try combining the pivots approach and see what comes out.

looked like a real goer on paper 😄

**Reply by DaveSkender** [reply]
I think for this concept, there’s several ways to get the info.  for me, I’d love to see some stats on how often this s a good and true signal

**Comment by sonutrades** [comment]
Hi @DaveSkender I am a new user of this library and really appreciate your efforts. I believe RSI/Price Divergence works and I would like to try it out. Can you please tell me how to get the divergence based on Price slope and RSI slope as mentioned above. Thanks

**Reply by DaveSkender** [reply]
Sadly, the conversation and the links above represent my total knowledge on this topic.  I’ve not spent any additional time thinking it through to a solution.
